### PR TITLE
Adding an organism-hints model

### PR DESCRIPTION
**Pull request by fileoy**
issue #1148 has many features regarding capturing hints in a paper.

This PR's main focus is on case number 3, which is **Species information is available**.

If the species information was available using PubTator 3.0, then certain information can be extracted.

### CODE REVIEW

File: src/model/hint.js [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+});
+const HINT_TYPES = _.flatMap(HINT_TYPE);
+
+
+const HINT_PASSAGE = Object.freeze({
+    TITLE: 'title',
+    ABSTRACT: 'abstract'
+});
+const HINT_PASSAGES = _.flatMap(HINT_PASSAGE);
+
```
Comment (jvwong) [inline src/model/hint.js:16]: Provide some brief documentation https://jsdoc.app/

File: src/model/hint.js [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+class Hint{
+
+    get text(){
```
Comment (jvwong) [inline src/model/hint.js:19]: What is this field: (a) the entire text (b) the mention (c) other?

File: src/model/hint.js [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+
+
+const HINT_PASSAGE = Object.freeze({
+    TITLE: 'title',
+    ABSTRACT: 'abstract'
+});
+const HINT_PASSAGES = _.flatMap(HINT_PASSAGE);
+
+class Hint{
+
```
Comment (jvwong) [inline src/model/hint.js:18]: Constructor?

**Review by jvwong — CHANGES_REQUESTED** [review]
Remove src/server/routes/api/document/hint/pubtator.js out of scope for this PR.

File: src/model/hint.js (hunk 2 above)
Comment (fileoy) [reply src/model/hint.js:19]: (b) the mention + added comments to clarify.

File: src/model/hint.js [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+const HINT_TYPES = _.flatMap(HINT_TYPE);
+
+const HINT_PASSAGE = Object.freeze({
+    TITLE: 'title',
+    ABSTRACT: 'abstract'
+});
+const HINT_PASSAGES = _.flatMap(HINT_PASSAGE);
+
+/**
+ * Class representing a Hint.
```
Comment (jvwong) [inline src/model/hint.js:15]: What is a Hint?

File: src/model/hint.js [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+    constructor(text, type, xref, section){
+        this._text = text;
+        this._type = type;
+        this._xref = xref;
+        this._section = section;
+    }
+
+    /**
+     * Get the hint text.
+     * @returns {string} The hint text.
```
Comment (jvwong) [inline src/model/hint.js:35]: What is 'text'? The whole paper? Some comment? Other?

**Review by jvwong — CHANGES_REQUESTED** [review]

**Review by jvwong — APPROVED** [review]
The text is the named entity mention.

**Pull request merged**